extmod/modnetwork.c

```diff
@@ -116,20 +116,20 @@ STATIC mp_obj_t network_country(size_t n_args, const mp_obj_t *args) {
 // TODO: Non-static to allow backwards-compatible pyb.country.
 MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(mod_network_country_obj, 0, 1, network_country);
 
-STATIC mp_obj_t network_hostname(size_t n_args, const mp_obj_t *args) {
+mp_obj_t mod_network_hostname(size_t n_args, const mp_obj_t *args) {
     if (n_args == 0) {
-        return mp_obj_new_str(mod_network_hostname, strlen(mod_network_hostname));
+        return mp_obj_new_str(mod_network_hostname_data, strlen(mod_network_hostname_data));
     } else {
         size_t len;
         const char *str = mp_obj_str_get_data(args[0], &len);
-        if (len >= MICROPY_PY_NETWORK_HOSTNAME_MAX_LEN) {
+        if (len > MICROPY_PY_NETWORK_HOSTNAME_MAX_LEN) {
             mp_raise_ValueError(NULL);
         }
-        strcpy(mod_network_hostname, str);
+        strcpy(mod_network_hostname_data, str);
```

Comment:
Would this be better as a `memcpy`, given that the length is already known, and to reduce dependence on null-terminated strings?

Reply:
Updated. It's not a direct switch because you also now need to explicitly null the end.

Reply:
It could do a memcpy with `len + 1`... that's guaranteed to work if strcpy was previously working :)

That would save a few bytes of code, and easy enough to change if/when we move away from null-terminated strings.